core/src/main/java/com/google/zxing/oned/OneDimensionalCodeWriter.java

```diff
@@ -33,8 +33,11 @@
 public abstract class OneDimensionalCodeWriter implements Writer {
   private static final Pattern NUMERIC = Pattern.compile("[0-9]+");
 
+  protected Map<EncodeHintType,?> hints;
```

Comment:
Rather than carry this around as a field, as it isn't otherwise used I believe, just change the implementation above - the current encode() method which does not accept hints can just call the version that does accept it, passing null, and then the current implementation becomes the body of an implementation that accepts the hints. This would be slightly more consistent with other impls.

Reply:
Yes, that's the better solution. I hope I solved it better now.